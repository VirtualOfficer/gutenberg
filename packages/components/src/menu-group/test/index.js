/**
 * External dependencies
 */
import { shallow } from 'enzyme';

/**
 * Internal dependencies
 */
import { MenuGroup } from '../';

describe( 'MenuGroup', () => {
	test( 'should render null when no children provided', () => {
		const wrapper = shallow( <MenuGroup /> );

		expect( wrapper.html() ).toBe( null );
	} );

	test( 'should match snapshot', () => {
		const wrapper = shallow(
			<MenuGroup label="My group" instanceId="1">
				<p>My item</p>
			</MenuGroup>
		);

		expect( wrapper ).toMatchSnapshot();
	} );

	test( 'should render given role', () => {
		const wrapper = shallow(
			<MenuGroup label="My group" instanceId="1" role="menu">
				<p>My item</p>
			</MenuGroup>
		);

		const menu = wrapper.find( '[role="menu"]' );
		expect( menu.length ).toBe( 1 );
	} );
} );
