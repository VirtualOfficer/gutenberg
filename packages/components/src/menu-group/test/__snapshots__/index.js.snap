// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MenuGroup should match snapshot 1`] = `
<div
  className="components-menu-group has-label"
>
  <div
    className="components-menu-group__label"
    id="components-menu-group-label-1"
  >
    My group
  </div>
  <NavigableMenu
    aria-labelledby="components-menu-group-label-1"
    orientation="vertical"
  >
    <p>
      My item
    </p>
  </NavigableMenu>
</div>
`;
